_source/_posts/2022-08-15-spring-boot-crud-update.md

```diff
@@ -0,0 +1,1077 @@
+---
+layout: blog_post
+title: "Spring Boot CRUD app with Vue and Quasar"
+author: andrew-hughes
+by: contractor
+communities: [java,javascript]
+description: "Build a Spring Boot Todo app with a Vue client using Okta for security and the Quasar framework for client components"
+tags: []
+tweets:
+- ""
+- ""
+- ""
+image:
+type: conversion
+
+---
+
+You're going to use Vue and Spring Boot to build a todo list web application. The application will include CRUD abilities, meaning that you will be able to **c**reate, **r**ead, **u**pdate, and **d**elete the todo items on the Spring Boot server via the client. The Vue front-end client will use the Quasar framework for presentation. Both the Spring Boot server and the Vue client will be secured with OAuth 2.0 and OpenID Connect (OIDC) using Okta as the security provider.
+
+{% img blog/spring-boot-crud-update/spring-and-vue.png alt:"Spring Boot, Vue, and Okta logos" width:"500" %}{: .center-image }
+
+This project has two major parts:
+
+- Spring Boot server
+- Vue client
+
+The Spring Boot server will include an H2 in-memory database and will use Spring Data JPA to map our todo data model to a database table for persistence. As you'll see, the server will leverage Spring Boot's ability to quickly expose data via a REST API with a minimum of configuration.
+
+The client will use [Vue 3](https://vuejs.org/) and the Quasar Framework. [The Quasar Framework](https://quasar.dev/) provides components and layout tools to help build Vue applications quickly with a consistent, high-quality user interface.
+
+**Prerequisites:**
+
+- [Java 11](https://adoptium.net/): or use [SDKMAN!](https://sdkman.io/) to manage and install multiple version
+
+- [Okta CLI](https://cli.okta.com/manual/#installation)
+
+- [HTTPie](https://httpie.org/doc#installation)
+
+- [Node 16+](https://nodejs.org)
+- [Vue CLI](https://cli.vuejs.org/guide/installation.html)
+
+You will need a free Okta Developer account if you don't already have one. But you can wait until later in the tutorial and use the Okta CLI to login or register for a new account.
+
+## Create Okta OIDC app
+
+Open a Bash shell. Create a parent directory for the project. Eventually, this will include both the resource server and client projects.
+
+```bash
+mkdir spring-boot-vue-crud
+cd spring-boot-vue-crud
+```
+
+If you already have an Okta account, log in using `okta login`. If you need to register for a new, free account: `okta register`.
+
+Create an OIDC app using the following command:
+
+```bash
+okta apps create
+```
+
+**Application name**: `spring-boot-vue-crud`
+
+**Type of Application**: `2: Single Page App`
+
+**Redirect URI(s)**: `http://localhost:8080/login/callback`
```

Comment:
You should be able to use the CLI include for this section. https://developer.okta.com/blog/setup#vue

Reply:
Does the CLI include allow me to specify a `redirect URI`?

Reply:
Yes. See the link above or the README for the syntax.